Update README.md (#8006)
Minio-io/minio: README.md

# MinIO Quickstart Guide
[![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io) [![Go Report Card](https://goreportcard.com/badge/minio/minio)](https://goreportcard.com/report/minio/minio) [![Docker Pulls](https://img.shields.io/docker/pulls/minio/minio.svg?maxAge=604800)](https://hub.docker.com/r/minio/minio/)

MinIO is an object storage server released under Apache License v2.0. It is compatible[1] with Amazon S3 cloud storage service. It is best suited for storing unstructured data such as photos, videos, log files, backups and container / VM images. Size of an object can range from a few KBs to a maximum of 5TB.

MinIO server is light enough to be bundled with the application stack, similar to NodeJS, Redis and MySQL.

> [1]: Some software such as S3QL will not work with MinIO in its default mode which is faster but not fully compatible with S3. For full compatibility with Amazon S3 protocols, start MinIO with `--compat` option.

Reviewer: ```suggestion
[1]: MinIO in its default mode is faster and does not calculate MD5Sum unless passed by client. This may lead to incompatibility with few S3 clients like s3ql that heavily depend on MD5Sum. For full compatibility with Amazon S3 API, start MinIO with `--compat` option.
```

Reviewer 2: We can also add that most applications don't need this requirement,  other than few specific ones likes s3ql which heavily depend on md5sum behavior.

Author: The problem with this is that the list of apps which require compatibility will always be incomplete since new apps are constantly being created. This is why I filed a feature request yesterday recommending that S3QL compatibility be the default. A much better approach is a list of apps which don't require compatibility as apps which don't require compatibility should still work even if you run in the compatible mode where the reverse is not true. Additionally, I would expect that the vast majority of users don't really need the higher performance, only the large scale server farm type users are likely to even care about being able to run it in a higher performance / incompatible mode. These are the same users which are most able to evaluate which applications can be run with or without compatibility.

FWIW.

Reviewer: > The problem with this is that the list of apps which require compatibility will always be incomplete since new apps are constantly being created. 

@SCDealy we don't need a list of compatible or incompatible applications here - that is too difficult to create and maintain. Regarding default behaviour will discuss if that is more favourable. But for the closure of this PR, if you could please take a look at the suggestion, it would be great.

Reviewer 2: MinIO use cases are driven by performance use cases, so the compat behavior is here to stay. The README.md change is sufficient for now.

> [1]: Some software such as S3QL will not work with MinIO in its default mode which is faster but not fully compatible with S3. For full compatibility with Amazon S3 protocols, it must be started in server (not gateway) mode and given the --compat option.

Reviewer 2: ```suggestion
[1]: Some software such as S3QL will not work with MinIO in its default mode which is faster but not fully compatible with S3. For full compatibility with Amazon S3 protocols, start MinIO with `--compat` option.
```

Author: Being new to MinIO (appx 1 day), I need some clarification. My original change was based on reading of past discussions / bug reports where it was stated (as I recall) that gateway mode was by design not fully S3 compatible. I don't know when the --compat option was added, but your change suggestion seems to imply that both server and gateway mode are compatible when --compat is selected. If so then I would agree with your suggested change.

Reviewer 2: Yes both gateway and server honor `--compat` option
